### PR TITLE
feat: implement session store with redis

### LINKED ISSUE #62: Implement session management using redis
Elasticache instance with Redis has already been provisioned on AWS

### PR DESCRIPTION
**Pull request by erickexinlu**
## Describe your changes
- wrote SessionService which provides functions for creating, validating, and accessing sessions using a string UUID
- removed deprecated JWT functionality in favor of redis session management
- updated Github Actions to include connections to docker redis instance

## Github Issue ID
Closes #62 

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: backend/src/app.go [hunk 1]
```diff
@@ -59,4 +63,5 @@ func (sa *ScheduliiApp) Run() error {
 
 func (sa *ScheduliiApp) Teardown() {
 	sa.db.Close()
+	// close redis?
```
Comment (lourw) [inline backend/src/app.go:66]: Yup

File: backend/src/db/redis.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"fmt"
+	"log"
+
+	"github.com/go-redis/redis"
+)
+
+func NewRedisConnection() (*redis.Client, error) {
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     "localhost:6379",
+		Password: "",
```
Comment (lourw) [inline backend/src/db/redis.go:13]: This stuff should be received from .env

File: backend/src/db/redis.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+		DB:       0,
+	})
+	PingRedis(rdb)
+	return rdb, nil
+}
+
+func PingRedis(client *redis.Client) {
```
Comment (lourw) [inline backend/src/db/redis.go:20]: If this doesn't do anything, I would remove. Just return an error if the client doesnt connect to anything

File: backend/src/repositories/session_repository.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+func NewSessionRepository(rdb *redis.Client) *SessionRepository {
+	return &SessionRepository{
+		rdb: rdb,
+	}
+}
+
+func (ss *SessionRepository) CreateSession(token *oauth2.Token, email string) (string, error) {
+	log.Println("EMAIL:" + email)
```
Comment (lourw) [inline backend/src/repositories/session_repository.go:25]: Remove log statements that arn't needed

File: backend/src/repositories/session_repository.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+		return "", err
+	}
+	return session.SessionId, ss.rdb.Set(session.SessionId, json, time.Hour).Err()
+}
+
+func (ss *SessionRepository) DeleteSession(sessionId string) error {
+	return ss.rdb.Del(sessionId).Err()
+}
+
+func (ss *SessionRepository) ValidateSession(sessionId string) (bool, error) {
```
Comment (lourw) [inline backend/src/repositories/session_repository.go:43]: Why do we need both a bool and an error?

File: backend/src/repositories/session_repository.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		return session, err
+	}
+	err = json.Unmarshal([]byte(val), &session)
+	if err != nil {
+		return session, err
+	}
+	return session, nil
+}
+
+func (ss *SessionRepository) GetOauthToken(sessionId string) (*oauth2.Token, error) {
```
Comment (lourw) [inline backend/src/repositories/session_repository.go:67]: Why do we need both GetOauthToken and GetSession?

File: backend/src/utils/google_oauth.go
Comment (lourw) [inline backend/src/utils/google_oauth.go:1]: I think the purpose of your session PR is to completely remove this util file.

File: backend/src/repositories/session_repository.go (hunk 5 above)
Comment (erickexinlu) [reply backend/src/repositories/session_repository.go:43]: at first i was thinking the behaviour for the redis being down or something should be different than if the session is invalid, but now that i think about it the end result is the same anyway

File: backend/src/repositories/session_repository.go (hunk 6 above)
Comment (erickexinlu) [reply backend/src/repositories/session_repository.go:67]: i thought maybe you would want to be able to access the other fields of the model, like the UserId that will connect it to our postgres database with the actual schedulii user's data, but i think it makes more sense to replace GetSession with GetScheduliiUUID or something like that

**Pull request merged**